### ISSUE TITLE
Clippy misses print macro usages when run as `cargo clippy`

### ISSUE THREAD

**Issue by danielrainer** [rust-lang/rust-clippy#14659]
### Summary

When the `print_stdout` and `print_stderr` lints are enabled, clippy does not warn about some usages of the corresponding macros, for example in tests.

Interestingly, this happens when running `cargo clippy` manually, but when running clippy via `rust-analyzer` from the editor, the lints show up as expected.

This has been discovered in the context of https://github.com/fish-shell/fish-shell/pull/11402.

### Lint Name

print_stdout, print_stderr

### Reproducer

I tried this code:

https://github.com/danielrainer/clippy_print_repro

`src/main.rs`:
```rust
fn main() {}

#[cfg(test)]
mod test {
    #[test]
    fn test() {
        println!("stdout");
        eprintln!("stderr");
    }
}
```

`Cargo.toml`:
```toml
[package]
name = "clippy_print_repro"
version = "0.1.0"
edition = "2024"

[dependencies]

[lints]
clippy.print_stdout = "warn"
clippy.print_stderr = "warn"
```

I expected to see this happen:

Warnings about using `println!` and `eprintln!` should be shown when running `cargo clippy`.

Instead, this happened:

No warning are shown when running `cargo clippy` (but they do show up through `rust-analyzer`).

### Version

```text
rustc 1.86.0 (05f9846f8 2025-03-31)
binary: rustc
commit-hash: 05f9846f893b09a1be1fc8560e33fc3c815cfecb
commit-date: 2025-03-31
host: x86_64-unknown-linux-gnu
release: 1.86.0
LLVM version: 19.1.7
```

**Comment by Alexendoo** [comment]
`cargo clippy` has the same default target selection behaviour as [`cargo check`](https://doc.rust-lang.org/cargo/commands/cargo-check.html#target-selection) which doesn't include tests/examples/benchmarks

**Comment by danielrainer** [comment]
Ah yes, thank you for pointing this out! Running `cargo clippy --workspace --all-targets` shows the lints as expected.
